parser_library/src/processing/instruction_sets/macro_processor.cpp

```diff
@@ -54,15 +55,35 @@ void macro_processor::process(std::shared_ptr<const processing::resolved_stateme
     auto args = get_args(*stmt);
     hlasm_ctx.enter_macro(stmt->opcode_ref().value, std::move(args.name_param), std::move(args.symbolic_params));
 }
-
-bool is_consuming_data_def(unsigned char c)
+namespace {
+bool is_valid_string(std::string_view s)
```

Comment:
- if guess that this checks whether `s` is a "generalized" string, right? Not just '...'?
- should we also consider the double apostrophe escape sequence?

Reply:
- You're right, it checks whether there is an even number of apostrophes. You can have e.g. the following:
```
      MACRO
      MAC
      ...
      MEND

      MAC SYM'THIS IS A STR''''''&VAR'
```
- Is the above example what you have in mind? Or are you thinking about something completely different?

Reply:
yes it is the example I am thinking of... and it is covered.